code/mob/living/critter/small_animal.dm

```diff
@@ -338,6 +339,385 @@ proc/filter_carrier_pets(var/type)
+	name = "turtle"
+	real_name = "turtle"
+	desc = "A turtle. They are noble creatures of the land and sea."
+	icon_state = "turtle"
+	icon_state_dead = "turtle-dead"
+	health_brute = 20
+	health_burn = 20
+	stamina = 0 // Turtles are slow
+	hand_count = 2
+	ai_retaliate_patience = 2
+	ai_retaliate_persistence = RETALIATE_UNTIL_INCAP
+	ai_type = /datum/aiHolder/turtle
+	player_can_spawn_with_pet = TRUE
+	density = FALSE
+	drop_contents_on_death = FALSE
+
+	var/shell_count = 0		//Count down to 0. Measured in process cycles. If they are in their shell when this is 0, exit.
+	var/rigged = FALSE
+	var/rigger = null
+	var/exploding = FALSE
+	var/costume_name = null
+	var/image/costume_alive = null
+	var/image/costume_shell = null
+	var/image/costume_dead = null
+
+	var/obj/item/wearing_beret = null
+	var/beret_remove_job_needed = null
+	var/list/allowed_hats = list(/obj/item/clothing/head/hos_hat, /obj/item/clothing/head/hosberet, new/obj/item/clothing/head/NTberet/commander)
+
+	add_abilities = list(/datum/targetable/critter/charge)
+	ai_attacks_per_ability = 0
+
+	New(loc)
+		. = ..()
+		START_TRACKING
+
+		#ifdef HALLOWEEN
+		var/r = rand(1,4)
+		costume_name = "sylv_costume_[r]"
+		#endif
+
+		if(!src.gender)
+			if(prob(50))
+				src.gender = MALE
+			else
+				src.gender = FEMALE
+
+		if (costume_name)
+			costume_alive = image(src.icon, "[costume_name]")
+			costume_shell = image(src.icon, "[costume_name]-shell")
+			costume_dead = image(src.icon, "[costume_name]-dead")
+
+	disposing()
+		. = ..()
+		STOP_TRACKING
+
+	get_desc()
+		..()
+		if (src.wearing_beret)
+			. += "<br>[src] is wearing an adorable beret!."
+		else
+			. += "<br>[src] looks cold without some sort of hat on."
+
+		if (src.costume_name)
+			. += "And he's wearing an adorable costume! Wow!"
+
+	update_icon()
+		if (isalive(src))
+			if (src.shell_count)
+				src.icon_state = "turtle-shell"
+			else if (src.wearing_beret)
+				if (istype(wearing_beret, /obj/item/clothing/head/hos_hat) || istype(wearing_beret, /obj/item/clothing/head/hosberet))
+					src.icon_state = "turtle-beret"
+				else if (istype(wearing_beret, /obj/item/clothing/head/NTberet/commander))
+					src.icon_state = "turtle-beret-com"
+			else
+				src.icon_state = "turtle"
+			if (costume_name)
+				src.UpdateOverlays(costume_alive, "costume")
+
+		else
+			if (src.wearing_beret)
+				if (istype(wearing_beret, /obj/item/clothing/head/hos_hat) || istype(wearing_beret, /obj/item/clothing/head/hosberet))
+					src.icon_state = "turtle-dead-beret"
+				else if (istype(wearing_beret, /obj/item/clothing/head/NTberet/commander))
+					src.icon_state = "turtle-dead-beret-com"
+			else
+				src.icon_state = "turtle-dead"
+			if (costume_name)
+				src.UpdateOverlays(costume_dead, "costume")
+
+	bullet_act(var/obj/projectile/P)
+		switch(P.proj_data.damage_type)
+			if(D_KINETIC,D_PIERCING,D_SLASHING)
+				if (prob(70))
+					src.enter_shell()
+		..()
+
+	attack_hand(mob/user)
+		if (isdead(src))
+			take_beret(user)
```

Comment:
This still doesn't actually work, you'll need to either allow the beret to be taken while the turtle is in their shell or make them exit their shell when they die.

Reply:
This was copypasted from old turtle code, im gonna remove it as you can still clickdrag